### PR TITLE
Add output_log_probability_x option

### PR DESCRIPTION
**Pull request by davidanthoff**
Fixes https://github.com/anthofflab/RobustAdaptiveMetropolisSampler.jl/issues/24.

Still needs a test.

@tonyewong, do you want to take this from here and add a test? You can just push ontop of this branch, I just gave you push access.

### CODE REVIEW

**Comment by tonyewong** [comment]
@davidanthoff added a couple of tests, and fixed (I think) an issue where the line adding to the vector of log-probabilities was giving an error and just needed some parentheses:
`output_log_probability_x && (log_probabilities_x[i] = log_probability_x)`

Want to check and merge?

File: test/runtests.jl [hunk 1]
```diff
@@ -9,21 +9,22 @@ using Test
...
+    result = RAM_sample(p -> logpdf(Normal(3., 2.), p[1]), [0.], [0.5], 1_000, show_progress=false, output_log_probability_x=false)
     @test length(result.chain) == 1_000
 
-    result = RAM_sample(p -> logpdf(Normal(3., 2.), p[1]), [0.], Diagonal([0.5]), 1_000, show_progress=false)
+    result = RAM_sample(p -> logpdf(Normal(3., 2.), p[1]), [0.], Diagonal([0.5]), 1_000, show_progress=false, output_log_probability_x=false)
     @test length(result.chain) == 1_000
-
-
-    result = RAM_sample(p -> logpdf(Normal(3., 2.), p[1]), [0.], fill(0.5, 1, 1), 1_000, show_progress=false)
+s
```
Comment (davidanthoff) [inline test/runtests.jl:26]: I think this `s` here is causing the tests to fail :)

Comment (tonyewong) [reply test/runtests.jl:26]: It's almost 6pm here! My fingers are so heavy from writing all 6 or however many of these lines of code...

**Comment by tonyewong** [comment]
Let's try that again...  🤦‍♂️

**Pull request merged**